### PR TITLE
Better bitvec reduction

### PR DESCRIPTION
**Pull request by alexkeizer**
### Description:

Relatively minor PR: adds some existing lemmas to the `minimal_theory` and `bitvec_rules` to get nicer normal forms. 
In particular:
- `BitVec.zero` now becomes `0#_`, and
- `truncate` now becomes `zeroExtend`, since the latter has simprocs for better reduction

Various tests have been changed accordingly

### Testing:

`make all` ran locally

### License:

By submitting this pull request, I confirm that my contribution is
made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: Proofs/Popcount32.lean [hunk 1]
```diff
@@ -136,11 +136,7 @@ info: popcount32_program.stepi_0x4005c0 (s sn : ArmState) (h_program : s.program
     (sn =
       w StateField.PC (4195780#64)
         (w (StateField.GPR 0#5)
-          (truncate 64 (BitVec.zero 32) &&& truncate 64 2147483648#32 |||
-            (truncate 64 (BitVec.zero 32) &&& 0#64 |||
-                truncate 64 ((zeroExtend 32 (r (StateField.GPR 0#5) s)).rotateRight 1) &&&
-                  truncate 64 4294967295#32) &&&
-              truncate 64 2147483647#32)
+          (zeroExtend 64 ((zeroExtend 32 (r (StateField.GPR 0#5) s)).rotateRight 1) &&& 4294967295#64 &&& 2147483647#64)
```
Comment (shigoel) [inline Proofs/Popcount32.lean:139]: Nice!

File: Arm/MinTheory.lean [hunk 2]
```diff
@@ -88,6 +88,9 @@ attribute [minimal_theory] bne_self_eq_false
 attribute [minimal_theory] bne_self_eq_false'
 attribute [minimal_theory] decide_False
 attribute [minimal_theory] decide_True
+attribute [minimal_theory]
+  decide_eq_false_iff_not
+  decide_eq_true_iff
```
Comment (shigoel) [inline Arm/MinTheory.lean:93]:
```suggestion
attribute [minimal_theory] decide_eq_false_iff_not
attribute [minimal_theory] decide_eq_true_iff
```
Stylistic conformance, I guess...

**Review by shigoel — COMMENTED** [review]
LGTM!

**Comment by alexkeizer** [comment]
@shigoel I accepted your comment!

(By the way, we would usually leave an "accept" review if something looks good except for a trivial suggestion, then I can hit merge myself after accepting said suggestion. That kind of workflow might save you some back-and-forth)

**Review by shigoel — APPROVED** [review]

**Pull request merged**